dChatServer/PlayerContainer.h

```diff
@@ -45,7 +43,7 @@ struct PlayerData {
 	std::string playerName;
 	std::vector<FriendData> friends;
 	std::vector<IgnoreData> ignoredPlayers;
-	eGameMasterLevel gmLevel;
+	eGameMasterLevel gmLevel = eGameMasterLevel::CIVILIAN;
```

Comment:
This was left out of the header on purpose to keep it small.  Is there a reason we must transitively include this header now?  Headers should be kept as small and have as few includes as possible in them, so changing this just adds another include for seemingly no reason?

Reply:
I wanted to avoid specifying a non-default constructor, even if said constructor only contains one assignment, and I figured the cost of #include-ing one enum class was worth it

Reply:
While I would normally agree, the pattern we are following is making headers as small as possible.  Not specifying a constructor is a want here rather than a need and I believe it is good practice to not include something you don't need.

Reply:
What about using a cast? That won't require an include

Reply:
Ended up using the cast

Reply:
I mean that _works_ but let me ask you this, in the future, if you saw `eGameMasterLevel gmLevel = static_cast<eGameMasterLevel>(0); // CIVILIAN`, I believe it would not be too far fetched to believe someone would want to remove the cast and just either include the file or make a constructor and do the work in the translation unit.  I believe it was fine before as a constructor as it initially was.  I know other would just say include the file, but my stance in headers has always been keep them small so transitive includes and rebuilds happen infrequently.  I won't block this change, but if I saw `eGameMasterLevel gmLevel = static_cast<eGameMasterLevel>(0); // CIVILIAN` I would just make a constructor and remove the comment and assignment.

Reply:
My reasoning is the moment you define a constructor - even if that's just to define it _as_ default - you open Pandora's box of all sorts of seemingly unrelated effects to copies, moves... Is it really worth it in this case solely to spare an include of a header file that only holds one single enum class?